src/ui/map.js

```diff
@@ -1341,6 +1341,34 @@ class Map extends Camera {
         return this.style.getLight();
     }
 
+    /**
+     * Sets state values on a feature, given a source and [optional]source-layer.
+     * 
+     * @param {string | {sourceId: string, sourceLayer: string}} source *For vector tile sources, the sourceLayer is
+     *  required.* For GeoJSON sources, just the source id.
+     * @param {string} feature The unique feature identifier from the source data.
+     * @param {string} key The name of the state to be set.
+     * @param {any} value The value to be set. This can be any valid JSON value.
+     */
+    setFeatureState(source: string | { sourceId: string; sourceLayer: string; }, feature: string, key: string, value: any) {
+        this.style.setFeatureState(source, feature, key, value);
+        this._update();
+    }
+
+    /**
+     * Gets the state of a feature, given a source and [optional]source-layer.
+     * 
+     * @param {string | {sourceId: string, sourceLayer: string}} source *For vector tile sources, the sourceLayer is
+     *  required.* For GeoJSON sources, just the source id.
+     * @param {string} feature The unique feature identifier from the source data.
+     * @param {string} key (optional) The name of the state. When null, all state values are returned.
+     * 
+     * @returns {any} The value of the specified specified state or all states.
+     */
+    getFeatureState(source: string | { sourceId: string; sourceLayer: string; }, feature: string, key?: string): any {
```

Comment:
The optional `key` param changes the return type. Should returning the full state object be split into a separate method? If it's useful to return the entire state as an object, should it be possible to set state with an object instead of individual calls for each key?